src/metrics/types/index.js

```diff
@@ -24,9 +24,29 @@ const Types = {
  */
 function getByName(name) {
 	/* istanbul ignore next */
-	if (!name) return null;
+	if (name === undefined) return null;
 
-	let n = Object.keys(Types).find(n => n.toLowerCase() == name.toLowerCase());
+	let converted = "Base";
+	switch (name) {
```

Comment:
It's a common code, not just Datadog specific, don't change it, because it will affect all reporters.